### PR TITLE
Railgun and Zone Railgun

### PR DESCRIPTION
**Pull request by nishanths**

### CODE REVIEW

File: railgun.go [hunk 1]
```diff
@@ -1,47 +1,311 @@
...
+}
+
+// railgunResponse represents the response from the Create Railgun and the Railgun Details endpoints.
+type railgunResponse struct {
+	Response
+	Result Railgun `json:"result"`
+}
+
+// railgunsResponse represents the response from the List Railguns endpoint.
+type railgunsResponse struct {
```
Comment (elithrar) [inline railgun.go]: `railgunListResponse` perhaps?

File: railgun.go [hunk 2]
```diff
@@ -1,47 +1,311 @@
...
+	ResponseStatus  string `json:"response_status"`
+	Protocol        string `json:"protocol"`
+	ElapsedTime     string `json:"elapsed_time"`
+	BodySize        string `json:"body_size"`
+	BodyHash        string `json:"body_hash"`
+	MissingHeaders  string `json:"missing_headers"`
+	ConnectionClose bool   `json:"connection_close"`
+	Cloudflare      string `json:"cloudflare"`
+	CFRay           string `json:"cf-ray"`
+	// XXX: CloudFlare's online API documentation does not yet have definitions
```
Comment (elithrar) [inline railgun.go]: s/XXX/NOTE

File: railgun.go [hunk 3]
```diff
@@ -1,47 +1,311 @@
...
+	// for the following fields. See: https://api.cloudflare.com/#railgun-connections-for-a-zone-test-railgun-connection/
+	CFWANError    interface{} `json:"cf-wan-error"`
```
Comment (elithrar) [inline railgun.go]: `string`

File: railgun.go [hunk 4]
```diff
@@ -1,47 +1,311 @@
...
+	CFCacheStatus interface{} `json:"cf-cache-status"`
```
Comment (elithrar) [inline railgun.go]: `string`

**Comment by elithrar** [comment]
Couple of minor comments, but otherwise looks good to me. 💯 for including tests.

File: railgun.go (hunk 1 above)
Comment (nishanths) [inline railgun.go]: Previously we've used the plural form for similar structs. Still change to `railgunListResponse`?

- https://github.com/cloudflare/cloudflare-go/blob/master/zone.go#L71
- https://github.com/cloudflare/cloudflare-go/blob/master/ssl.go#L33

**Comment by nishanths** [comment]
@elithrar Addressed the comments!

File: railgun.go (hunk 1 above)
Comment (elithrar) [inline railgun.go]: Good point. It's not a major issue either way since it's private—we can change it later if need be!

**Comment by elithrar** [comment]
LGTM. @jamesog and/or @jamesog to review and we can merge!

**Pull request merged**